waiter/src/waiter/shell_scheduler.clj

```diff
@@ -358,42 +358,62 @@
     instance))
 
 (defn- get-pid->memory
-  "Issues and parses the results of a ps shell command and returns a new pid->memory map"
+  "Issues and parses the results of a ps shell command and returns a pid->memory fn."
   []
   (try
-    (let [ps-output (-> (sh/sh "ps" "-eo" "pid,pgid,rss") :out (str/split #"\n") rest)
-          pid->pgid (map #(-> % str/trim (str/split #"\s+") butlast
-                              ((fn [[a b]] [(Integer/parseInt a) (Integer/parseInt b)])))
-                         ps-output)
-          pgid->rss (apply merge-with +
-                           (map #(-> % str/trim (str/split #"\s+") rest
-                                     ((fn [[a b]] {(Integer/parseInt a) (Integer/parseInt b)})))
-                                ps-output))]
-      (into {} (for [[pid pgid] pid->pgid] [pid (get pgid->rss pgid)])))
+    (let [cols->entry (fn [[pid ppid rss]] {:pid (Integer/parseInt pid)
```

Comment:
We can use `zipmap` here: `(fn [elements] (zipmap [: pid :ppid :rss] elements)`

Reply:
Where were you planning on putting the `Integer/parseInt` calls?

Reply:
`(zipmap [:pid :ppid :rss] (map Integer/parseInt elements))` :)

Reply:
The original is clearer and doesn't try to take abuse the coincidence that every value is an `int`.  If we ever need a string we have have to revert to something like this.